x/feegrant/types/msgs.go

```diff
@@ -1,22 +1,40 @@
 package types
 
 import (
+	"time"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/feegrant/exported"
 )
 
-// MsgGrantFeeAllowance adds permission for Grantee to spend up to Allowance
-// of fees from the account of Granter.
-// If there was already an existing grant, this overwrites it.
-type MsgGrantFeeAllowance struct {
-	Granter   sdk.AccAddress        `json:"granter" yaml:"granter"`
-	Grantee   sdk.AccAddress        `json:"grantee" yaml:"grantee"`
-	Allowance exported.FeeAllowance `json:"allowance" yaml:"allowance"`
+func (msg MsgGrantFeeAllowance) NewMsgGrantFeeAllowance(feeAllowanceI exported.FeeAllowance, granter, grantee sdk.AccAddress) (MsgGrantFeeAllowance, error) {
+	feeallowance := &FeeAllowance{}
```

Comment:
This is not application-specific, right?

Reply:
This needs little cleanup, it should be same as before. Will update